File: ts/src/test/Exchange/test.fetchWithdrawals.ts
import assert from 'assert';
import { Exchange } from "../../../ccxt";
import testDepositWithdrawal from './base/test.depositWithdrawal.js';
import testSharedMethods from './base/test.sharedMethods.js';

async function testFetchWithdrawals (exchange: Exchange, skippedProperties: string[], code: string) {
    const method = 'fetchWithdrawals';
    const transactions = await exchange.fetchWithdrawals (code);
    testSharedMethods.assertNonEmtpyArray (exchange, skippedProperties, method, transactions, code);
    const now = exchange.milliseconds ();
    for (let i = 0; i < transactions.length; i++) {
        testDepositWithdrawal (exchange, skippedProperties, method, transactions[i], code, now);
    }
    testSharedMethods.assertTimestampOrder (exchange, method, code, transactions);
}

export default testFetchWithdrawals;

File: julia/Ccxt/test/exchange/test.fetchWithdrawals.jl
using Test
using Ccxt
function testFetchWithdrawals(exchange, skippedProperties, code)

    method = "fetchWithdrawals";
    transactions = Base.fetch(fetchWithdrawals(exchange, code));
    assertNonEmtpyArray(testSharedMethods, exchange, skippedProperties, method, transactions, code);
    now = milliseconds(exchange);
    i = 0
    while functions.ccxtruthy(functions.ccxt_lt(i, length(transactions)))
        testDepositWithdrawal(exchange, skippedProperties, method, get(transactions, i + 1, nothing), code, now);
        i += 1
    end
    assertTimestampOrder(testSharedMethods, exchange, method, code, transactions);
    return true
end
